programs/eosio-launcher/main.cpp

```diff
@@ -1271,6 +1304,12 @@ launcher_def::write_bios_boot () {
          string key = line.substr(len);
          if (key == "envars") {
             brb << "bioshost=" << bhost << "\nbiosport=" << biosport << "\n";
+            char cluster_str[20];
+            sprintf(cluster_str, "cluster%05d", cluster_id);
+            brb << "wddir=$wddir/" << (const char *)cluster_str << "\n";
+            brb << "wdaddr=localhost:" << calculate_port(cluster_id, 0, port_type::keosd_boot) << "\n";
+            brb << "wdurl=http://$wdaddr\n";
+            brb << "wpidfile=ignition_wallet" << cluster_str << ".pid\n";
```

Comment:
This could say, for example, "wpidfile=ignition_walletcluster00000". Did you mean to add a _ to separate them?

Better yet I think this line should be:
```cpp
brb << "wpidfile=$wddir/ignition_wallet.pid\n";
```

That way all these files are kept together in the appropriate cluster subdirectory.

In fact, at that point, both this line and the one above it can just be kept in the testnet.template file after the `###INSERT envars` line.